File: api/schemas/groups.py
from pydantic import BaseModel
from typing import List, Literal, Optional


class GroupCreate(BaseModel):
    name: str
    type: Optional[Literal["category", "slider"]] = None
    slider_min: Optional[int] = None
    slider_max: Optional[int] = None
    slider_labels: Optional[List[str]] = None


class GroupUpdate(BaseModel):
    name: Optional[str] = None
    color: Optional[str] = None
    icon: Optional[str] = None
    sort_order: Optional[int] = None
    type: Optional[Literal["category", "slider"]] = None
    slider_min: Optional[int] = None
    slider_max: Optional[int] = None
    slider_labels: Optional[List[str]] = None


class ReorderRequest(BaseModel):
    ordered_ids: List[int]


class OptionCreate(BaseModel):
    name: str


class OptionUpdate(BaseModel):
    name: Optional[str] = None
    icon: Optional[str] = None
    sort_order: Optional[int] = None

File: src/shared/schemas/groups.ts
import { z } from 'zod';

export const groupTypeSchema = z.enum(['category', 'slider']);

export const groupCreateSchema = z.object({
  name: z.string().min(1),
  type: groupTypeSchema.optional(),
  slider_min: z.number().int().optional(),
  slider_max: z.number().int().optional(),
  slider_labels: z.array(z.string()).optional(),
});

export const groupUpdateSchema = z.object({
  name: z.string().min(1).optional(),
  color: z.string().optional(),
  icon: z.string().optional(),
  sort_order: z.number().int().optional(),
  type: groupTypeSchema.optional(),
  slider_min: z.number().int().optional(),
  slider_max: z.number().int().optional(),
  slider_labels: z.array(z.string()).nullable().optional(),
});

export const reorderRequestSchema = z.object({
  ordered_ids: z.array(z.number().int()),
});

export const optionCreateSchema = z.object({
  name: z.string().min(1),
});

export const optionUpdateSchema = z.object({
  name: z.string().min(1).optional(),
  icon: z.string().optional(),
  sort_order: z.number().int().optional(),
});

export type GroupCreateInput = z.infer<typeof groupCreateSchema>;
export type GroupUpdateInput = z.infer<typeof groupUpdateSchema>;
export type ReorderRequestInput = z.infer<typeof reorderRequestSchema>;
export type OptionCreateInput = z.infer<typeof optionCreateSchema>;
export type OptionUpdateInput = z.infer<typeof optionUpdateSchema>;
